AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Models/AWSCognitoNetworkPreferences.swift

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+public struct AWSCognitoNetworkPreferences {
+
+    /// The maximum number of retries for failed requests. The value needs to be between 0 and 10 inclusive.
+    /// If set to higher than 10, it becomes 10.
+    public let maxRetryCount: UInt32
+
+    /// The timeout interval to use when waiting for additional data.
+    public let timeoutIntervalForRequest: Double
```

Comment:
should this just be `TimeInterval` type since that's always specified in seconds? If it's just a Double, would suggest something that says "seconds" in there `timeoutRequestInSeconds`

Reply:
`TimeInterval` **is** a `Double` anyway, so I vote to change the type to be explicit on what we expect (i.e. seconds).

```swift
typealias TimeInterval = Double
```

Reply:
As @ruisebas  mentioned, `Double` to `TimeInterval` is non-breaking change since it's just a typealias.

No naming changes, TimeInterval is expected to be in seconds, other places have names like "backgroundTrackingTimeout", "timeout", "sessionRefreshInterval", etc.

Amplify core provides a public extension for TimeInterval: https://github.com/aws-amplify/amplify-swift/blob/main/Amplify/Core/Support/TimeInterval%2BHelper.swift This seems useful if we changed the type to TimeInterval.

If we agree on above, looks like we can just do it now since it's non-breaking